### PR TITLE
Add Artifactory API to get package artifact lead file

### PR DESCRIPTION
**Pull request by oshratZairi**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

Adding API to retrieve the package lead file from Artifactory.

### CODE REVIEW

File: artifactory/services/packages.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		return nil, err
+	}
+	return body, errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK)
+}
+
+type LeadFileRequest struct {
+	PackageVersion  string `json:"package_version"`
+	PackageName     string `json:"package_name"`
+	PackageRepoName string `json:"package_repo_name"`
+	PackageType     string `json:"package_type"`
```
Comment (yahavi) [inline artifactory/services/packages.go:50]: Please consider adding "omitempty"

**Review by yahavi — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1438,6 +1439,11 @@ serviceManager.FileList("repo/path/", optionalParams)
 ```go
 storageInfo, err := serviceManager.GetStorageInfo()
 ```
+#### Getting Package Artifact LeadFile
```
Comment (yahavi) [inline README.md:1442]:
```suggestion
#### Getting Package Artifact Lead File
```

File: artifactory/emptymanager.go [hunk 3]
```diff
@@ -114,6 +115,10 @@ type ArtifactoryServicesManager interface {
 type EmptyArtifactoryServicesManager struct {
 }
 
+func (esm *EmptyArtifactoryServicesManager) GetPackageLeadFile(leadFileRequest services.LeadFileRequest) ([]byte, error) {
+	panic("Failed: Method is not implemented")
+}
+
```
Comment (yahavi) [inline artifactory/emptymanager.go:121]: Let's move it to below ImportReleaseBundle, to keep the same order

File: artifactory/services/packages.go (hunk 1 above)
Comment (oshratZairi) [reply artifactory/services/packages.go:50]: all fields are mandatory

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**